### DISCUSSION TITLE
Fundamentos C# - Versões

### DISCUSSION THREAD

**Discussion by Lorenzogrocha** [balta-io/forum#602] [Tira Dúvidas] [answered]
Olá. Li a respeito que pretendia refazer o curso com a versão mais recente. Seja dotnet 8 ou 9. Existe uma previsão? Comecei recentemente e infelizmente a gente acaba enrolando um pouco devido erros ou configurações que foram alteradas devido as versões. Muita coisa está diferente hoje em dia (até o arquivos Program já vem com uma sintaxe diferente no código, pra um iniciante isso atrapalha), estou indeciso se desinstalo e instalo a versão da aula, ou se continuo na versão mais recente. Mas como disse, meio que perde tempo as vezes com configurações.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Lorenzogrocha, como vai?

Recomendamos sempre utilizar as mesmas versões do curso. O .NET roda lado-a-lado, como explico nas aulas iniciais do curso, então não tem problema instalar a versão 3, 5, 7... juntas.

Caso queira usar a versão mais nova (8), é plenamente possível, basta utilizar a flag `--use-program-main` na hora de criar o projeto, assim o código virá igual era até a versão 6 do .NET:
```
dotnet new console -o MeuApp --use-program-main
```

**Comment by Lorenzogrocha** [comment]
Show, irei prosseguir desta maneira. Obrigado pela atenção 🙏
